pulp_ansible/app/downloaders.py

```diff
@@ -51,25 +52,31 @@ def __init__(
         self.silence_errors_for_response_status_codes = silence_errors_for_response_status_codes
         super().__init__(url, **kwargs)
 
-    def raise_for_status(self, response):
+    def raise_for_status(self, response, handle_401=False):
         """
         Raise error if aiohttp response status is >= 400 and not silenced.
 
+        Args:
+            handle_401(bool): If true, catch 401, request a new token and retry.
+
         Raises:
             FileNotFoundError: If aiohttp response status is 404 and silenced.
             aiohttp.ClientResponseError: If the response status is 400 or higher and not silenced.
 
         """
         silenced = response.status in self.silence_errors_for_response_status_codes
 
-        if not silenced:
+        if handle_401 and response.status == 401:
+            self.registry_auth["bearer"] = None
+
+        elif not silenced:
             response.raise_for_status()
 
         if response.status == 404:
             raise FileNotFoundError()
 
     @backoff.on_exception(backoff.expo, ClientResponseError, max_tries=10, giveup=http_giveup)
-    async def _run(self, extra_data=None):
+    async def _run(self, handle_401=True, extra_data=None):
```

Comment:
Is there any place where the code calls with `handle_401=False`? Unless this param is in use I'd suggest to have this be the way the downloader works until configurability is needed.

Reply:
Oh I see it is used further down...